### PR TITLE
CMSSW_10_2_X LLP filter - backporting  #37669

### PR DESCRIPTION
**Pull request by Raffaella07**
#### PR description:
The filter generalizes MCLongLivedParticles GenFilter (filter was included in GenFilters up to CMSSW_11_0_X/ , for CMSSW_10_2_X see https://github.com/cms-sw/cmssw/blob/CMSSW_10_2_X/GeneratorInterface/GenFilters/src/MCLongLivedParticles.cc
It selects particles based on transverse displacement (by means of a lower and/or upper cut) and pdgIds.
The filter is developed for searches of long lived HNLs in B meson decays (@rmanzoni @pandolf @amlyon)

The filter is included in the `plugins` subdirectory of `GeneratorInterface/GenFilters` .
It is possible to retrieve the original filter behavior by setting the lower cut only and leaving the pdgIds unspecified.
NOTE: the filter syntax has changed with respect to the original version, as a lower (LengMin) and an upper (LengMax) cut can be specified, while in the CMSSW_10_2_X/  version there was only a lower cut parameter.

PR validation:
Filter behavior can be tested in the `test` subdirectory of `GeneratorInterface/GenFilters` through the module `test_MCLongLivedFilter_cfg.py`

Backporting  #37669

### CODE REVIEW

File: SimG4Core/Generators/src/Generator.cc [hunk 1]
```diff
@@ -33,7 +33,8 @@ Generator::Generator(const ParameterSet & p) :
   theMinPCut(p.getParameter<double>("MinPCut")),    // in GeV (CMS standard)
   theMaxPCut(p.getParameter<double>("MaxPCut")),   
   theEtaCutForHector(p.getParameter<double>("EtaCutForHector")),
-  verbose(p.getUntrackedParameter<int>("Verbosity",0)),
+  //verbose(p.getUntrackedParameter<int>("Verbosity",0)),
```
Comment (civanch) [inline SimG4Core/Generators/src/Generator.cc:36]: @Raffaella07 , please, revert back previous line.

File: SimG4Core/Generators/src/Generator.cc [hunk 2]
```diff
@@ -106,6 +107,46 @@ void Generator::HepMC2G4(const HepMC::GenEvent * evt_orig, G4Event * g4evt)
 
   HepMC::GenEvent *evt=new HepMC::GenEvent(*evt_orig);
 
+
```
Comment (civanch) [inline SimG4Core/Generators/src/Generator.cc:110]: @Raffaella07 , please revert back

**Comment by civanch** [comment]
-1

backport assuming taking code from the master and adopting to the previous release. In #37669 only 3 files was changed. Here some code for debugging  is peaked up, which is not in the master.

**Comment by Raffaella07** [comment]
Sorry I accidentally pushed from a developing branch which was way ahead of CMSSW_10_2_X, I'll close this and produce a PR with the new files only. Sorry for the spam.

**Pull request closed without merging**